internal/controllers/topology/cluster/desired_state.go

```diff
@@ -696,13 +707,23 @@ func computeMachineDeployment(ctx context.Context, s *scope.Scope, machineDeploy
 		return nil, errors.Wrap(err, "failed to calculate desired infrastructure machine template ref")
 	}
 
+	nameTemplate := "{{ .cluster.name }}-{{ .machineDeployment.topologyName }}-{{ .random }}"
+	if machineDeploymentClass.NamingStrategy != nil && machineDeploymentClass.NamingStrategy.Template != nil {
+		nameTemplate = *machineDeploymentClass.NamingStrategy.Template
+	}
+
+	name, err := names.MachineDeploymentNameGenerator(nameTemplate, s.Current.Cluster.Name, machineDeploymentTopology.Name).GenerateName()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to generate name for machine deployment")
```

Comment:
```suggestion
		return nil, errors.Wrap(err, "failed to generate name for MachineDeployment")
```
I know it's a bit arbitrary how we name things in errors around here but let's try to use the actual names if we talk about core CAPI CRDs (same for MP below)


Let's also make sure we have consistent error wrapping for all generate cases (~ we should add errors.Wrap to  cp, bootstraptemplate, ...)